go/vt/vtgate/engine/primitive.go

```diff
@@ -72,7 +72,10 @@ type VCursor interface {
 	// Resolver methods, from key.Destination to srvtopo.ResolvedShard.
 	// Will replace all of the Topo functions.
 	ResolveDestinations(keyspace string, ids []*querypb.Value, destinations []key.Destination) ([]*srvtopo.ResolvedShard, [][]*querypb.Value, error)
+
 	SetTarget(target string) error
+	SetUDV(key string, value interface{}) error
```

Comment:
I don't think we should add more to this interface. We should do the refactoring we've talked about instead.